api/src/pcapi/routes/backoffice/chronicles/blueprint.py

```diff
@@ -148,4 +224,76 @@ def unpublish_chronicle(chronicle_id: int) -> utils.BackofficeResponse:
         chronicle=chronicle,
     )
     flash(f"La chronique {chronicle_id} à été dépubliée", "success")
-    return redirect(url_for("backoffice_web.chronicles.list_chronicles"), code=303)
+    return redirect(request.referrer or url_for("backoffice_web.chronicles.list_chronicles"), code=303)
+
+
+@chronicles_blueprint.route("/<int:chronicle_id>/attach-product", methods=["POST"])
+@atomic()
+@permission_required(perm_models.Permissions.MANAGE_CHRONICLE)
+def attach_product(chronicle_id: int) -> utils.BackofficeResponse:
+    if not FeatureToggle.WIP_ENABLE_CHRONICLES_IN_BO.is_active():
+        raise NotFound()
+
+    form = forms.AttachProductForm()
+    if not form.validate():
+        mark_transaction_as_invalid()
+        flash(utils.build_form_error_msg(form), "warning")
+        redirect(url_for("backoffice_web.chronicles.details", chronicle_id=chronicle_id), code=303)
+
+    chronicle = chronicles_models.Chronicle.query.get_or_404(chronicle_id)
+    product = offers_models.Product.query.filter(offers_models.Product.extraData["ean"].astext == form.ean.data).first()
+
+    if not product:
+        mark_transaction_as_invalid()
+        flash("Aucune œuvre n'à été trouvée pour cet EAN")
+        return redirect(url_for("backoffice_web.chronicles.details", chronicle_id=chronicle_id), code=303)
+
+    chronicle.products.append(product)
+    db.session.add(product)
+    db.session.flush()
+
+    flash(f"{product.name} a été rattaché à la chronique", "success")
```

Comment:
Pour que ce soit plus homogène, mais aussi pour éviter l'injection XSS dans un nom de produit : 
```suggestion
    flash(Markup("Le produit <b>{product_name}</b> a été rattaché à la chronique").format(product_name=product.name), "success")
```
Car en testant en local, avec un script dans le nom du produit je vois cela : 
![image](https://github.com/user-attachments/assets/b163feac-be27-4c2f-8466-d43d5b7f2b51)
![image](https://github.com/user-attachments/assets/e8ad70e8-3039-4815-8c90-65e74e5e913c)